src/parson.c

```diff
@@ -488,10 +490,16 @@ static JSON_Status json_object_dotremove_internal(JSON_Object *object, const cha
     JSON_Object *temp_object = NULL;
     char *dot_pos = NULL;
     int name_len = 0;
+    char *name2 = NULL;
 
     /* SAFEC modified to use _s */
     name_len = strnlen_s(name, STRING_NAME_MAX);
-    strstr_s((char *)name, name_len, ".", 1, &dot_pos);
+    name2 = calloc(STRING_NAME_MAX, sizeof(char));
```

Comment:
Unsure I follow, so just double checking:
- is name2 ever filled with data? seems to be an empty buffer for its lifespan
-Is it possible to lose a char of data if name_len = STRING_NAME_MAX?  (aka should calloc size be STRING_NAME_MAX + 1)

Reply:
Good catch, didn't cut/paste the strncpy_s to each instance, will add that and re-test.